### PR TITLE
[SwatchColorPicker] add a renderfunction as props to override cell content

### PR DESCRIPTION
**Pull request by pKalaga**
## Previous Behavior

SwatchColorPicker takes in a onRenderCell in props, which overrides the entire button/cell in the grid. This is not ideal if consuming code just needs to override svg or the cell content, and not the button. Passing in classes/styles also doesnot satisfy this requirement completely, if you need to provide custom styling for each button.

## New Behavior
Proposal here is to expose a new property which just overrides the cell content and keeps the button behavior intact.
For example, if a GridPicker is needed for setting backgrounds or gradients, this can help customize button content, but retain onClick/onFocus/onHover behaviors.

## Related Issue(s)

- Fixes  # #28065

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/17ca96c6b7f684f36a5c63e40e78b4986a86c713'>17ca96c6b7f684f36a5c63e40e78b4986a86c713</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv8"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/BaseButton-mount_base.html">635</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/BaseButton-mount.html">610</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Breadcrumb-mount_base.html">1659</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Breadcrumb-mount.html">1679</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Checkbox-mount_base.html">1685</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Checkbox-mount.html">1680</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/CheckboxBase-mount_base.html">1455</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/CheckboxBase-mount.html">1490</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/ChoiceGroup-mount_base.html">2878</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/ChoiceGroup-mount.html">2905</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/ComboBox-mount_base.html">644</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/ComboBox-mount.html">655</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/CommandBar-mount_base.html">6303</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/CommandBar-mount.html">6183</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/ContextualMenu-mount_base.html">12313</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/ContextualMenu-mount.html">12113</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/DefaultButton-mount_base.html">749</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/DefaultButton-mount.html">767</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/DetailsRow-mount_base.html">2146</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/DetailsRow-mount.html">2124</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/DetailsRowFast-mount_base.html">2181</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/DetailsRowFast-mount.html">2186</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/DetailsRowNoStyles-mount_base.html">1989</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/DetailsRowNoStyles-mount.html">2002</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Dialog-mount_base.html">2604</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Dialog-mount.html">2587</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/DocumentCardTitle-mount_base.html">215</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/DocumentCardTitle-mount.html">239</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Dropdown-mount_base.html">1965</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Dropdown-mount.html">1941</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/FocusTrapZone-mount_base.html">1131</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/FocusTrapZone-mount.html">1080</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/FocusZone-mount_base.html">1048</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/FocusZone-mount.html">1066</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/GroupedList-mount_base.html">41114</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/GroupedList-mount.html">41569</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/GroupedList-virtual-rerender_base.html">19845</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/GroupedList-virtual-rerender.html">19840</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/GroupedList-virtual-rerender-with-unmount_base.html">50529</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/GroupedList-virtual-rerender-with-unmount.html">50620</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/GroupedListV2-mount_base.html">227</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/GroupedListV2-mount.html">229</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/GroupedListV2-virtual-rerender_base.html">218</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/GroupedListV2-virtual-rerender.html">205</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/GroupedListV2-virtual-rerender-with-unmount_base.html">225</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/GroupedListV2-virtual-rerender-with-unmount.html">234</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/IconButton-mount_base.html">1071</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/IconButton-mount.html">1062</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Label-mount_base.html">331</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Label-mount.html">328</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Layer-mount_base.html">2690</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Layer-mount.html">2688</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Link-mount_base.html">398</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Link-mount.html">392</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/MenuButton-mount_base.html">936</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/MenuButton-mount.html">944</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/MessageBar-mount_base.html">21160</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/MessageBar-mount.html">21290</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Nav-mount_base.html">1930</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Nav-mount.html">1917</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/OverflowSet-mount_base.html">755</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/OverflowSet-mount.html">763</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Panel-mount_base.html">1730</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Panel-mount.html">1764</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Persona-mount_base.html">739</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Persona-mount.html">740</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Pivot-mount_base.html">868</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Pivot-mount.html">874</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/PrimaryButton-mount_base.html">844</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/PrimaryButton-mount.html">843</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Rating-mount_base.html">4597</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Rating-mount.html">4566</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/SearchBox-mount_base.html">879</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/SearchBox-mount.html">906</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Shimmer-mount_base.html">1884</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Shimmer-mount.html">1837</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Slider-mount_base.html">1313</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Slider-mount.html">1324</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/SpinButton-mount_base.html">2848</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/SpinButton-mount.html">2814</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Spinner-mount_base.html">390</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Spinner-mount.html">400</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/SplitButton-mount_base.html">1800</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/SplitButton-mount.html">1855</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Stack-mount_base.html">404</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Stack-mount.html">404</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/StackWithIntrinsicChildren-mount_base.html">841</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/StackWithIntrinsicChildren-mount.html">854</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/StackWithTextChildren-mount_base.html">2590</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/StackWithTextChildren-mount.html">2636</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/SwatchColorPicker-mount_base.html">6008</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/SwatchColorPicker-mount.html">6113</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/TagPicker-mount_base.html">1430</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/TagPicker-mount.html">1427</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Text-mount_base.html">380</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Text-mount.html">363</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/TextField-mount_base.html">910</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/TextField-mount.html">910</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/ThemeProvider-mount_base.html">836</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/ThemeProvider-mount.html">823</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/ThemeProvider-virtual-rerender_base.html">591</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/ThemeProvider-virtual-rerender.html">585</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1272</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1266</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Toggle-mount_base.html">608</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/Toggle-mount.html">608</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/buttonNative-mount_base.html">182</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28064/perf-test/buttonNative-mount.html">193</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/components/SwatchColorPicker/SwatchColorPicker.types.ts [hunk 1]
```diff
@@ -88,6 +88,11 @@ export interface ISwatchColorPickerProps extends React.RefAttributes<HTMLElement
    */
   onRenderColorCell?: IRenderFunction<IColorCellProps>;
 
+  /**
+   * Custom render function for button/cell content area
+   */
+  onRenderColorCellContent?: IRenderFunction<IColorCellProps>;
```
Comment (micahgodbolt) [inline packages/react/src/components/SwatchColorPicker/SwatchColorPicker.types.ts:94]: I'm also curious why there is onRenderColorCellContent and onRenderCellContent

2 asks:
1. Write an example of how these new props would be used to customize the control (either in PR description, or by creating a new website example). I'm curious if you need to use both onRenders or just one.
2. explain the reasoning for the names being different. and make it obvious to a user which onRender they should be worried about.

**Review by micahgodbolt — CHANGES_REQUESTED** [review]
2 asks:

1. Write an example of how these new props would be used to customize the control (either in PR description, or by creating a new website example). I'm curious if you need to use both onRenders or just one.
2. explain the reasoning for the two onRender names being different. and make it obvious to a user which onRender they should be worried about.

File: packages/react/src/components/SwatchColorPicker/SwatchColorPicker.types.ts (hunk 1 above)
Comment (pKalaga) [reply packages/react/src/components/SwatchColorPicker/SwatchColorPicker.types.ts:94]: Will rename these to have the same name `onRenderColorCellContent`. 
These are on two different components, one on SwatchColorPicker, which calls `ColorPickerGridCell`, so users generally need to only pass one prop to SwatchColorPicker.

**Comment by pKalaga** [comment]
@microsoft/cxe-red , @micahgodbolt - have added sample code and updated api as per comments, need another review here

File: packages/react/src/components/SwatchColorPicker/SwatchColorPicker.types.ts (hunk 1 above)
Comment (micahgodbolt) [reply packages/react/src/components/SwatchColorPicker/SwatchColorPicker.types.ts:96]: Okay, that makes sense. I like them better as the same name. Makes it obvious that the parent is just passing this same named prop to all of its children. The approach we typically take in newer v9 is to use context to pass the values down, but with v8 it isn't required due to parent manually rendering the children.

Thanks for the demo, looks great!

**Review by micahgodbolt — APPROVED** [review]

**Comment by pKalaga** [comment]
@micahgodbolt - can these changes be merged? is there anything else i am missing?

**Review by micahgodbolt — APPROVED** [review]

**Pull request merged**